misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/ProgramAnalyzerNodeVisitor.java

```diff
@@ -375,1133 +196,165 @@ public void visit(FunctionDefinitionNode functionDefinitionNode) {
-
-    }
-
-    @Override
-    public void visit(XMLFilterExpressionNode xMLFilterExpressionNode) {
-
-    }
-
-    @Override
-    public void visit(XMLStepExpressionNode xMLStepExpressionNode) {
-
-    }
-
-    @Override
-    public void visit(XMLNamePatternChainingNode xMLNamePatternChainingNode) {
-
-    }
-
-    @Override
-    public void visit(XMLAtomicNamePatternNode xMLAtomicNamePatternNode) {
-
-    }
-
-    @Override
-    public void visit(TypeReferenceTypeDescNode typeReferenceTypeDescNode) {
-
-    }
-
-    @Override
-    public void visit(MatchStatementNode matchStatementNode) {
-
-    }
-
-    @Override
-    public void visit(MatchClauseNode matchClauseNode) {
-
-    }
-
-    @Override
-    public void visit(MatchGuardNode matchGuardNode) {
-
-    }
-
-    @Override
-    public void visit(DistinctTypeDescriptorNode distinctTypeDescriptorNode) {
-
-    }
-
-    @Override
-    public void visit(ListMatchPatternNode listMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(RestMatchPatternNode restMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(MappingMatchPatternNode mappingMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(FieldMatchPatternNode fieldMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(ErrorMatchPatternNode errorMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(NamedArgMatchPatternNode namedArgMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(MarkdownDocumentationNode markdownDocumentationNode) {
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
 
-    }
+        Optional<Symbol> symbol = model.symbol(variableDeclarationNode);
+        if (symbol.isEmpty()) {
+            return;
+        }
 
-    @Override
-    public void visit(MarkdownDocumentationLineNode markdownDocumentationLineNode) {
+        if (variableDeclarationNode.initializer().isEmpty()) {
+            return;
+        }
 
-    }
+        if (isClientObject(symbol.get())) {
+            resolveEndPoint(symbol.get(), variableDeclarationNode.initializer().get());
+        }
 
-    @Override
-    public void visit(MarkdownParameterDocumentationLineNode markdownParameterDocumentationLineNode) {
+        if (isClientObject(symbol.get()) || isRecordObject(symbol.get())) {
+            Optional<ExpressionNode> expressionNode = variableDeclarationNode.initializer();
 
-    }
+            if (expressionNode.isPresent() && expressionNode.get().kind() != SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                this.registerVariableRef(getEndpointReference(symbol.get()), variableDeclarationNode);
+            }
+        } else if ((symbol.get().kind() == SymbolKind.VARIABLE) &&
+                (variableDeclarationNode.initializer().get().kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION)) {
+            this.registerVariableRef(variableDeclarationNode.typedBindingPattern().bindingPattern().
```

Comment:
Why do we use the `hashCode` here? If you are going to use this against any sort of equality check, please use the object itself. Have  a look at the `equals` method of the `LineRange` class. It has been overridden

Reply:
The hash code here is used as the unique identifier. An unique identified is required in the choreo API.

Reply:
hashCode() cannot be considered as unique (that's why we have hash collisions in hash maps). If `id` can be a `string` what if we use `lineRange.toString() `instead?

Reply:
Use a UUID mapping with the line range. But I'm not sure why we need to uniquely identify the line range since it inherently unique.

Reply:
Fixed